tracer_options.go

```diff
@@ -47,6 +49,21 @@ func (tracerOptions) Logger(logger Logger) TracerOption {
 	}
 }
 
+func (tracerOptions) CustomHeaderKeys(headerKeys *HeadersConfig) TracerOption {
```

Comment:
this is now a tracer option. If custom header keys are provided in the configuration, new codecs are created and added to the tracers injectors/extractors.